server/models/Itinerary.js

```diff
@@ -0,0 +1,34 @@
+const mongoose = require('mongoose');
+const { Schema } = mongoose;
+const Cars = require("./Cars")
+const Flights = require("./Flights")
+const Hotels = require("./Hotel")
+const User = require("./User")
+
+const itinerarySchema = new Schema({
+  user: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: User,
+    required: true,
+  },
+  car: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: Cars,
+    required: false,
+  },
+  flight: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: Flights,
+    required: false,
+  },
```

Comment:
maybe a good idea here would be to use a [subDocument](https://mongoosejs.com/docs/subdocs.html) so that we don't have to add the car/flight/hotel to our database, but we can keep that information embedded in the itinerary. This is a combination of what you had in mind with the nested objects, but more structured and better practice